extension/docs/time_weighted_average.md

```diff
@@ -0,0 +1,254 @@
+# Time Weighted Average [<sup><mark>experimental</mark></sup>](/extension/docs/README.md#tag-notes)
+
+> [Description](#time-weighted-average-description)<br>
+> [API](#time-weighted-average-api) <br>
+> [Sample Usage](time-weighted-average-examples)<br>
+> [Interpolation Methods Details](#time-weight-methods)<br>
+> [Notes on Parallelism and Ordering](#time-weight-ordering)<br>
+
+## Description [](time-weighted-average-description)
+
+Time weighted averages are commonly used in cases where a time series is not evenly sampled, so a traditional average will give misleading results. Consider a voltage sensor that sends readings once every 5 minutes or whenever the value changes by more than 1 V from the previous reading. If the results are generally stable, but with some quick moving transients, a simple average over all of the points will tend to over-weight the transients instead of the stable readings. A time weighted average weights each value by the duration over which it occured based on the points around it and produces correct results for unevenly spaced series. 
+
+Timescale Analytics' time weighted average is implemented as an aggregate which weights each value either using a last observation carried forward (LOCF) approach or a linear interpolation approach ([see interpolation methods](#time-weight-methods)). While the aggregate is not parallelizable, it is supported with [continuous aggregation](https://docs.timescale.com/latest/using-timescaledb/continuous-aggregates). 
+
+Additionally, see the notes on 
+
+
+## Command List (A-Z) [](time-weighted-average-api)
+> - [time_weight() (point form)](#time_weight_point)
+> - [time_weight() (summary form)](#time-weight-summary) 
+> - [average()](#time-weight-average)
+
+---
+## **time_weight() (point form)** [](time_weight_point)
+```SQL
+timescale_analytics_experimental.time_weight(
+    method TEXT¹,
+    ts TIMESTAMPTZ,
+    value DOUBLE PRECISION
+) RETURNS TimeWeightSummary
+```
+¹ Only two values are currently supported, 'linear' and 'LOCF', any capitalization of these will be accepted. [See interpolation methods for more info.](#time-weight-methods) 
+
+An aggregate that produces a `TimeWeightSummary` from timestamps and associated values. 
+
+### Required Arguments² [](time-weight-point-required-arguments)
+|Name| Type |Description|
+|---|---|---|
+| `method` | `TEXT` | The weighting method we should use, options are 'linear' or 'LOCF', not case sensitive |
+| `ts` | `TIMESTAMPTZ` |  The time at each point |
+| `value` | `DOUBLE PRECISION` | The value at each point to use for the time weighted average|
+<br>
+
+² Note that `ts` and `value` can be `null`, however the aggregate is not evaluated on `null` values and will return `null`, but it will not error on `null` inputs.
+
+### Returns
+
+|Column|Type|Description|
+|---|---|---|
+| `time_weight` | `TimeWeightSummary` | A TimeWeightSummary object that can be passed to other functions within the time weighting API. |
+<br>
```

Comment:
I do think that having a short example for each command in-line in the documentation (in addition to the following section) is useful.

Reply:
hmmm...yeah. I think that's probably true, it's just that they're kinda interlinked here, but I'll give it a shot.